lib/demos/material/data_table_demo.dart

```diff
@@ -17,11 +17,39 @@ class DataTableDemo extends StatefulWidget {
   _DataTableDemoState createState() => _DataTableDemoState();
 }
 
+class _RestorableSelectedDessertIndices extends RestorableValue<Set<int>> {
```

Comment:
This design makes it tempting to just modify the `values` set directly, which doesn't work because those updates will not be restored.

Instead of exposing the value set directly, I would write a custom RestorableProperty, which does not expose the set and instead has an interface like this:

```
class Foo extends ... {
  setSelection(Set<int> selectedIndexes);
  isSelected(int index);
}
```

Reply:
Done